### ISSUE TITLE
Saving before sleeping caused game to crash and corrupt save!!

### ISSUE THREAD

**Issue by Zetsukaze** [CleverRaven/Cataclysm-DDA#31105]
# Describe the bug

As stated in title

# Steps To Reproduce

1. Try to sleep
2. Pick save before sleeping
3. Game crashes

# Screenshots

# Versions and configuration

 - OS: Windows 10
 - Game Version: 0.D-3880-g89feb2b (tiles)

# Additional context

Debug log:
```
00:34:20.525 ERROR SDL : (error message will follow backtrace)
	(@0x59764e[cataclysm-tiles.exe+0x19764e]), 
	(@0x597c7a[cataclysm-tiles.exe+0x197c7a]), 
	(@0xcc078c[cataclysm-tiles.exe+0x8c078c]), 
	(@0xce4ac8[cataclysm-tiles.exe+0x8e4ac8]), 
	(@0x658235[cataclysm-tiles.exe+0x258235]), 
	(@0x6adf79[cataclysm-tiles.exe+0x2adf79]), 
	(@0x496b79[cataclysm-tiles.exe+0x96b79]), 
	(@0x6c96ae[cataclysm-tiles.exe+0x2c96ae]), 
	(@0x6aa8ba[cataclysm-tiles.exe+0x2aa8ba]), 
	(IMG_LoadWEBP_RW+0x5314e8@0x14c8808[cataclysm-tiles.exe+0x10c8808]), 
	(@0x4013ed[cataclysm-tiles.exe+0x13ed]), 
	(@0x4014fb[cataclysm-tiles.exe+0x14fb]), 
	(BaseThreadInitThunk+0x14@0x7ffc8f6a7974[KERNEL32.DLL+0x17974]), 
	(RtlUserThreadStart+0x21@0x7ffc9049a271[ntdll.dll+0x6a271]), 
Backtrace emission took 0 seconds.
src/sdlsound.cpp:508: Failed to play sound effect: Tried to play a chunk with a bad frame
00:35:21.207 ERROR SDL : (error message will follow backtrace)
	(@0x59764e[cataclysm-tiles.exe+0x19764e]), 
	(@0x597c7a[cataclysm-tiles.exe+0x197c7a]), 
	(@0xcc078c[cataclysm-tiles.exe+0x8c078c]), 
	(@0xce4ac8[cataclysm-tiles.exe+0x8e4ac8]), 
	(@0x658235[cataclysm-tiles.exe+0x258235]), 
	(@0x6adf79[cataclysm-tiles.exe+0x2adf79]), 
	(@0x496b79[cataclysm-tiles.exe+0x96b79]), 
	(@0x6c96ae[cataclysm-tiles.exe+0x2c96ae]), 
	(@0x6aa8ba[cataclysm-tiles.exe+0x2aa8ba]), 
	(IMG_LoadWEBP_RW+0x5314e8@0x14c8808[cataclysm-tiles.exe+0x10c8808]), 
	(@0x4013ed[cataclysm-tiles.exe+0x13ed]), 
	(@0x4014fb[cataclysm-tiles.exe+0x14fb]), 
	(BaseThreadInitThunk+0x14@0x7ffc8f6a7974[KERNEL32.DLL+0x17974]), 
	(RtlUserThreadStart+0x21@0x7ffc9049a271[ntdll.dll+0x6a271]), 
Backtrace emission took 0 seconds.
src/sdlsound.cpp:508: Failed to play sound effect: Tried to play a chunk with a bad frame
00:52:03.871 ERROR SDL : (error message will follow backtrace)
	(@0x59764e[cataclysm-tiles.exe+0x19764e]), 
	(@0x597c7a[cataclysm-tiles.exe+0x197c7a]), 
	(@0xcc078c[cataclysm-tiles.exe+0x8c078c]), 
	(@0xce4ac8[cataclysm-tiles.exe+0x8e4ac8]), 
	(@0x658235[cataclysm-tiles.exe+0x258235]), 
	(@0x6adf79[cataclysm-tiles.exe+0x2adf79]), 
	(@0x496b79[cataclysm-tiles.exe+0x96b79]), 
	(@0x6c96ae[cataclysm-tiles.exe+0x2c96ae]), 
	(@0x6aa8ba[cataclysm-tiles.exe+0x2aa8ba]), 
	(IMG_LoadWEBP_RW+0x5314e8@0x14c8808[cataclysm-tiles.exe+0x10c8808]), 
	(@0x4013ed[cataclysm-tiles.exe+0x13ed]), 
	(@0x4014fb[cataclysm-tiles.exe+0x14fb]), 
	(BaseThreadInitThunk+0x14@0x7ffc8f6a7974[KERNEL32.DLL+0x17974]), 
	(RtlUserThreadStart+0x21@0x7ffc9049a271[ntdll.dll+0x6a271]), 
Backtrace emission took 0 seconds.
src/sdlsound.cpp:508: Failed to play sound effect: Tried to play a chunk with a bad frame
00:58:27.394 ERROR SDL : (error message will follow backtrace)
	(@0x59764e[cataclysm-tiles.exe+0x19764e]), 
	(@0x597c7a[cataclysm-tiles.exe+0x197c7a]), 
	(@0xcc078c[cataclysm-tiles.exe+0x8c078c]), 
	(@0xce4ac8[cataclysm-tiles.exe+0x8e4ac8]), 
	(@0x658235[cataclysm-tiles.exe+0x258235]), 
	(@0x6adf79[cataclysm-tiles.exe+0x2adf79]), 
	(@0x496b79[cataclysm-tiles.exe+0x96b79]), 
	(@0x6c96ae[cataclysm-tiles.exe+0x2c96ae]), 
	(@0x6aa8ba[cataclysm-tiles.exe+0x2aa8ba]), 
	(IMG_LoadWEBP_RW+0x5314e8@0x14c8808[cataclysm-tiles.exe+0x10c8808]), 
	(@0x4013ed[cataclysm-tiles.exe+0x13ed]), 
	(@0x4014fb[cataclysm-tiles.exe+0x14fb]), 
	(BaseThreadInitThunk+0x14@0x7ffc8f6a7974[KERNEL32.DLL+0x17974]), 
	(RtlUserThreadStart+0x21@0x7ffc9049a271[ntdll.dll+0x6a271]), 
Backtrace emission took 0 seconds.
```

EDIT 1:
Sorry, not sure if the above  error message from the debug log is the crash. 
Another crash occured without the debug log stuff from above.
Steps:
1. Setting zones in a new basecamp. 

Logs from crash.log
```
CRASH LOG FILE: config/crash.log
VERSION: 0.D-3880-g89feb2b
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:
    @0x582F15[cataclysm-tiles.exe+0x182F15]
    @0x5839E2[cataclysm-tiles.exe+0x1839E2]
    SMPEG_error+0x4B034@0xEA54B0[cataclysm-tiles.exe+0xAA54B0]
    _C_specific_handler+0x98@0x7FFC8FD18068[msvcrt.dll+0x28068]
    _chkstk+0x11F@0x7FFC904D477F[ntdll.dll+0xA477F]
    RtlWalkFrameChain+0x14BF@0x7FFC90434BEF[ntdll.dll+0x4BEF]
    KiUserExceptionDispatcher+0x2E@0x7FFC904D34EE[ntdll.dll+0xA34EE]
    IMG_LoadWEBP_RW+0x42575C@0x13BCA7C[cataclysm-tiles.exe+0xFBCA7C]
    @0x97A2B6[cataclysm-tiles.exe+0x57A2B6]
    @0x6E5B98[cataclysm-tiles.exe+0x2E5B98]
    @0x6ABD77[cataclysm-tiles.exe+0x2ABD77]
    @0x6AC17E[cataclysm-tiles.exe+0x2AC17E]
    @0x6CADF2[cataclysm-tiles.exe+0x2CADF2]
    @0x6AA8BA[cataclysm-tiles.exe+0x2AA8BA]
    IMG_LoadWEBP_RW+0x5314E8@0x14C8808[cataclysm-tiles.exe+0x10C8808]
    @0x4013ED[cataclysm-tiles.exe+0x13ED]
    @0x4014FB[cataclysm-tiles.exe+0x14FB]
    BaseThreadInitThunk+0x14@0x7FFC8F6A7974[KERNEL32.DLL+0x17974]
    RtlUserThreadStart+0x21@0x7FFC9049A271[ntdll.dll+0x6A271]
```

**Comment by Zetsukaze** [comment]
Save from edit 1 is attached, just interact with the bulletin board.
[Crash.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3249385/Crash.zip)

**Comment by ZhilkinSerg** [comment]
Probably fixed with #31123.
